src/pages/my_page/user_profile/UserOrcid.tsx

```diff
@@ -61,29 +94,31 @@ export const UserOrcid = ({ user }: UserOrcidProps) => {
           method: 'PATCH',
           data: { orcid },
         });
-        if (isSuccessStatus(addOrcidResponse.status)) {
-          dispatch(setNotification({ message: t('feedback.success.update_orcid'), variant: 'success' }));
-          cristinPersonQuery.refetch();
-        } else if (isErrorStatus(addOrcidResponse.status)) {
-          dispatch(setNotification({ message: t('feedback.error.update_orcid'), variant: 'success' }));
+        const orcidCredentials = getOrcidCredentials(history.location.search, orcidInfoResponse.data.id);
+        if (!orcidCredentials) {
+          dispatch(setNotification({ message: t('feedback.error.storing_orcid_credentials'), variant: 'warning' }));
+        } else {
+          const postOrcidCredentialsResponse = await postOrcidCredentials(orcidCredentials);
+          if (isSuccessStatus(addOrcidResponse.status) && isSuccessStatus(postOrcidCredentialsResponse.status)) {
+            dispatch(setNotification({ message: t('feedback.success.update_orcid'), variant: 'success' }));
+          } else if (isErrorStatus(addOrcidResponse.status) || isErrorStatus(postOrcidCredentialsResponse.status)) {
+            dispatch(setNotification({ message: t('feedback.error.update_orcid'), variant: 'success' }));
```

Comment:
Denne skulle hatt error status
```suggestion
            dispatch(setNotification({ message: t('feedback.error.update_orcid'), variant: 'error' }));
```
Men det blir egneltig heller ikke helt rett da, for hvis `addOrcidResponse` blir OK så har jo brukeren fått til det man ønsket, og vi vil vise ORCID-tilkoblingen etterpå, mens det er det skjulte ekstrakallet som kanskje feilet.
Man vil altså få feilmeledingen `Oppdatering av ORCID-kobling feilet`, for så å se at Orcid-koblingen ligger der :p

Reply:
må tenke litt selv på denne merker jeg 🤔  (etter lunsj)

Reply:
Litt ekkel denne altså, si det blir 3 operasjoner som brukeren bare oppfatter som 1. For brukeren vil man tro at enten er handlingen vellykket eller mislykket, men den kan være 0/3 success, 1/3  success, 2/3  success eller 3/3  success, så vanskelig å si hvordan man skal informerer brukeren om ting. Tror det kan være lurt å gjøre dette sekvensielt, og avbryte alt gjenværende med en gang noe blir feil. Da blir det mer synlig at det feila, og at brukeren må prøve på nytt. Tror det blir litt feil og noen smutthull uansett hvordan vi gjør det da :/

Hvis vi gjør noe ala det under her så tror jeg det vil gi mening for brukeren å prøve på nytt uansett hva som feiler. Virker det logisk? 🤔 
```
if (isSuccessStatus(addOrcidResponse.status)) {
  dispatch(setNotification({ message: t('feedback.success.update_orcid'), variant: 'success' }));
  const orcidCredentials = getOrcidCredentials(history.location.search, orcidInfoResponse.data.id);
  if (!orcidCredentials) {
    dispatch(setNotification({ message: t('feedback.error.storing_orcid_credentials'), variant: 'error' }));
  } else {
    const postOrcidCredentialsResponse = await postOrcidCredentials(orcidCredentials);
    if (isErrorStatus(postOrcidCredentialsResponse.status)) {
      dispatch(setNotification({ message: t('feedback.error.storing_orcid_credentials'), variant: 'error' }));
    }
  }
} else if (isErrorStatus(addOrcidResponse.status)) {
  dispatch(setNotification({ message: t('feedback.error.update_orcid'), variant: 'error' }));
}
```

Reply:
Litt ekkel denne altså, si det blir 3 operasjoner som brukeren bare oppfatter som 1. For brukeren vil man tro at enten er handlingen vellykket eller mislykket, men den kan være 0/3 success, 1/3  success, 2/3  success eller 3/3  success, så vanskelig å si hvordan man skal informerer brukeren om ting. Tror det kan være lurt å gjøre dette sekvensielt, og avbryte alt gjenværende med en gang noe blir feil. Da blir det mer synlig at det feila, og at brukeren må prøve på nytt. Tror det blir litt feil og noen smutthull uansett hvordan vi gjør det da :/

Hvis vi gjør noe ala det under her så tror jeg det vil gi mening for brukeren å prøve på nytt uansett hva som feiler. Virker det logisk? 🤔 
```
if (isSuccessStatus(addOrcidResponse.status)) {
  dispatch(setNotification({ message: t('feedback.success.update_orcid'), variant: 'success' }));
  const orcidCredentials = getOrcidCredentials(history.location.search, orcidInfoResponse.data.id);
  if (!orcidCredentials) {
    dispatch(setNotification({ message: t('feedback.error.storing_orcid_credentials'), variant: 'error' }));
  } else {
    const postOrcidCredentialsResponse = await postOrcidCredentials(orcidCredentials);
    if (isErrorStatus(postOrcidCredentialsResponse.status)) {
      dispatch(setNotification({ message: t('feedback.error.storing_orcid_credentials'), variant: 'error' }));
    }
  }
} else if (isErrorStatus(addOrcidResponse.status)) {
  dispatch(setNotification({ message: t('feedback.error.update_orcid'), variant: 'error' }));
}
```

Reply:
^ett problem med den der igjen er når bare den siste POSTen av orcid data feiler, siden brukeren da vil først få en OK-melding tett etterfulgt av en feilmelding. Men jeg tror det er the lesser of many evils

Reply:
Det vil ikke være noen måte å prøve på nytt igjen om bare den siste feiler da, siden brukeren brukeren da vil ha en tilkoblet ORCID. Så kanskje gir det mening å bytte om slik at det blir til slutt (etter lagring av ORCID-data)

Reply:
Etter at orcid har blitt lagret på brukeren, er det i praksis ikke mulig å forsøke én gang til.

Reply:
Nettopp. Kanskje best å prøve å lagre orcid-data før vi prøver å koble orcid til brukeren da (og avbryte alt når noe feiler), slik at brukeren alltid for lov til å prøve på nytt hvis noe går feil

Reply:
Det går ikke an. Da handleren sjekker om at du forsøker å lagre orcid credentials som er dine egne.
Hvis du forsøker å lagre orcid credentials og du ikke har de så vil du få en forbidden message.

Reply:
Det beste hadde jo vært at hele greia er en backend task, det var det jeg begynte på, men fikk beskjed om at versjon 1 er å gjøre minst mulig i code-reviewsene mine.

Reply:
hmm nettopp, så da er den utelukket ja 😅 
Står igjen med casen hvor en bruker får koblet til orcid-en da, men lagring av data kan feile, og brukeren er da "låst" til det med mindre man fjerner orcid for så å legge til igjen, som man ikke har noe insentiv for å gjøre. Men det er kanskje ikke så farlig om noen evnt mangler den dataen? Alle som har gjort dette før vil uansett havne rett i den staten så.
Men høres ut som lagring av orcid-data burde være i success-handler til `addOrcidResponse` da, siden vi vet at det vil feile ellers

Reply:
Endret til varianten du foreslo lengre opp. Den gir jo da som du sier muligens to litt motsigende meldinger for samme lastinga.